src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -736,6 +787,63 @@ private bool ArePointerTypesEqual(PointerTypeSymbol type, PointerTypeSymbol othe
                 return AreTypesEqual(type.PointedAtType, other.PointedAtType);
             }
 
+            private bool AreFunctionPointerTypesEqual(FunctionPointerTypeSymbol type, FunctionPointerTypeSymbol other)
+            {
+                var sig = type.Signature;
+                var otherSig = other.Signature;
+
+                if (!isEqual(sig.ReturnTypeWithAnnotations, otherSig.ReturnTypeWithAnnotations,
+                             sig.RefKind, otherSig.RefKind,
+                             sig.RefCustomModifiers, otherSig.RefCustomModifiers,
+                             allowOut: false))
+                {
+                    return false;
+                }
+
+                if (sig.ParameterCount != otherSig.ParameterCount)
+                {
+                    return false;
+                }
+
+                for (int i = 0; i < sig.ParameterCount; i++)
+                {
+                    var param = sig.Parameters[i];
+                    var otherParam = otherSig.Parameters[i];
+                    if (!isEqual(param.TypeWithAnnotations, otherParam.TypeWithAnnotations,
+                                param.RefKind, otherParam.RefKind,
+                                param.RefCustomModifiers, otherParam.RefCustomModifiers,
+                                allowOut: true))
+                    {
+                        return false;
+                    }
+                }
+
+                return true;
+
+                bool isEqual(TypeWithAnnotations type, TypeWithAnnotations otherType, RefKind refKind, RefKind otherRefKind, ImmutableArray<CustomModifier> refCustomModifiers, ImmutableArray<CustomModifier> otherRefCustomModifiers, bool allowOut)
```

Comment:
>bool [](start = 16, length = 4)

Can we separate the assertion logic from the actual functionality? Seems like it would be more readable to have `isValidFunctionPointerParameter(...)` function and then call `Debug.Assert(isValidFunctionPointerParameter(...))` separately from `refKind != otherRefKind || !AreTypesEqual(type.Type, otherType.Type)`

Reply:
The only issue with that is, until we have attributes on local functions available for use, I can't put `Conditional("DEBUG")` on such a local function, so we'll emit a call to an empty function.

Reply:
You can make it a static method :)